xapian-core/tests/api_sorting.cc

```diff
@@ -319,3 +319,49 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
     );
     return true;
 }
+
+DEFINE_TESTCASE(replace_weights, backend) {
+    Xapian::Database mydb(get_database("apitest_onedoc"));
+    Xapian::Enquire enquire(mydb);
+    enquire.set_query(Xapian::Query("word"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    // old_max_possible = 0.269763689697702
+    double old_max_possible = mymset.get_max_possible();
+    double new_weight = 0.25;
+    vector<double> weights {new_weight};
```

Comment:
You don't need the extensibility of a vector here, so using `std::array` is better: http://en.cppreference.com/w/cpp/container/array

```
array<double, 1> weights = {{ new_weight }};
```